content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -682,6 +682,190 @@ The new Job itself will have a different uid from `a8f3d00d-c6d2-11e5-9f87-42010
+      - name: main
+        image: main-image:v1
+        ...
+  backoffLimit: 6
+  podFailurePolicy:
+    rules:
+    - action: FailJob
+      onExitCodes:
+        containerName: main      # optional
+        operator: In             # one of: In, NotIn
+        values: [42]
+    - action: Ignore             # one of: Ignore, FailJob, Count
+      onPodConditions:
+      - type: DisruptionTarget   # indicates Pod disruption
+```
+
+The first rule of the Pod failure policy, in the example above, specifies that
+the Job should be marked failed if the `main` container fails with the 42 exit
+code. For the `main` container specifically:
+
+- an exit code of 0 means that the container succeeded
+- an exit code of 42 means that the **entire Job** failed
+- any other exit code represents that the container failed, and hence the entire
+  Pod. The Pod will be re-created if the total number of restarts is
+  below `backoffLimit`. If the `backoffLimit` is reached the **entire Job** failed.
+
+{{< note >}}
+Because the Pod template specifies a `restartPolicy: Never`,
+the kubelet does not restart the `main` container in that particular Pod.
+{{< /note >}}
+
+The second rule of the Pod failure policy, specifying the `Ignore` action for
+failed Pods with condition `DisruptionTarget` excludes Pod disruptions from
+being counted towards the `.spec.backoffLimit` counter.
+
+{{< note >}}
+If the Job failed, either by the Pod failure policy or Pod backoff
+failure policy, and the Job is running multiple Pods, Kubernetes shuts down all
+the Pods in that Job.
+{{< /note >}}
+
+These are some requirements and semantics of the API:
+- if you want to use a `.spec.podFailurePolicy` field for a Job, you must
+  also define that Job's pod template with `.spec.restartPolicy` set to `Never`.
+- the Pod failure policy rules you specify under `spec.podFailurePolicy.rules`
+  are evaluated in order. Once a rule matches a Pod failure, the remaining of
+  the rules are ignored. When no rule matches the Pod failure, the default
+  handling applies.
+- you may want to restrict a rule to a specific container by specifing its name
+  in`spec.podFailurePolicy.rules[*].containerName`. When not specified the rule
+  applies to all containers. When specified, it should match one the container
+  or initContainer names in the Pod template.
+- you may specify the action taken when a Pod failure policy is matched by
+  `spec.podFailurePolicy.rules[*].action`. Possible values are:
+  - `FailJob`: use to indicate that the Pod's job should be marked as Failed and
+     all running Pods should be terminated.
+  - `Ignore`: use to indicate that the counter towards the `.spec.backoffLimit`
+     should not be incremented and a replacement Pod should be created.
+  - `Count`: use to indicate that the Pod should be handled in the default way -
+     the counter towards the `.spec.backoffLimit` should be incremented.
+
+#### Pod failure policy examples {#pod-failure-policy-examples}
```

Comment:
This kind of content doesn't fit in "concepts".

If you consider this is important, you could create a new task https://kubernetes.io/docs/tasks/job/

Reply:
My intention for this section is to improve understanding of the concept of "Pod failure policy". I think providing a short example helps with this goal. This approach is used for example for init containers https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#using-init-containers. Similarly for `Deployments` https://kubernetes.io/docs/concepts/workloads/controllers/deployment/.
But I agree the long section might be overloading the user with examples. 

I guess we can package the examples as a "Reducing costs using Pod failure policy" task or such.

Reply:
Thanks @mimowo